tests/suites/storage/persistent_storage.sh

```diff
@@ -0,0 +1,145 @@
+# This function tests charm deployment with a single filesystem storage unit and a single persistent block device
+# storage unit.
+#  Steps taken to the test:
+#       - Deploy dummy-storage charm with a single block storage unit (ebs)
+#         and a single filesystem storage unit (rootfs).
+#       - Check charm status once the deployment is done.
+#           > Application status should be active.
+#       - Check charm storage units once the deployment is done.
+#           > Total number of storage units should be 2.
+#           > Name of storage units should be in align with charm config.
+#           > Properties of storage units should be as defined.
+#               - Storage Type, Persistent Setting and Pool.
+run_persistent_storage() {
+	echo
+
+	model_name="persistent-storage"
+	file="${TEST_DIR}/test-${model_name}.log"
+	ensure "${model_name}" "${file}"
+	# dummy-storage is going to be deployed with 1 ebs block storage unit
+	# and 1 rootfs filesystem storage unit.
+	echo "dummy-storage is going to be deployed with 1 ebs block storage unit and 1 rootfs filesystem storage unit"
+	juju --debug deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-blk=ebs \
+		--storage single-fs=rootfs
+	# wait for current application-status to be active
+	wait_for "active" '.applications | .[] | ."application-status" | .current'
+	# wait for current workload-status to be active
+	wait_for "active" "$(workload_status "dummy-storage" 0).current"
+	echo "Checking current status of app dummy-storage."
+	juju status --format json | jq -S '.applications | .[] | ."application-status" | .current' | check "active"
+
+	echo "Checking total number of storage unit(s)."
+	juju storage --format json | jq '.storage | keys | length' | check 2
+	echo "Checking names of storage unit(s)."
+	juju storage --format json | jq '.storage | keys | .[0]' | check "single-blk/0"
+	juju storage --format json | jq '.storage | keys | .[1]' | check "single-fs/1"
+	echo "Check name and total number of storage unit: PASSED."
+
+	#
+	# check type, persistent setting and pool of single block storage unit
+	#
+	# storage type
+	juju storage --format json | jq '.storage | ."single-blk/0" | .kind' | check "block"
```

Comment:
list-storage is an alias for storage. You should probably choose one and stick with it for consistency